### PR TITLE
popup upgrade pro dialog component is created

### PR DESCRIPTION
**Pull request by adilkadivala**
solved #24

When you click the upgrade to pro button, it should open the dialog.

### CODE REVIEW

File: src/components/navigation/Navigation.tsx [hunk 1]
```diff
@@ -35,10 +37,14 @@ export const Navigation = () => {
...
-          <button className="flex items-center justify-center w-full gap-1 py-2 font-semibold text-indigo-500 border rounded-full hover:text-black hover:border-gray-300">
+          <Button
+            variant="outline"
+            className="flex items-center justify-center w-full gap-1 py-2 font-semibold text-indigo-500 border rounded-full hover:text-black hover:border-gray-300"
+            onClick={handleDialog}
+          >
             <BoltIcon className="w-4 h-4" />
             Upgrade to Pro
-          </button>
+          </Button>
```
Comment (bryanlundberg) [inline src/components/navigation/Navigation.tsx:47]: Here you should directly insert the dialog logic (sample on shadcn dialogs section)
And only create a new component for the dialog content.

File: src/components/navigation/Pricing.tsx
Comment (bryanlundberg) [inline src/components/navigation/Pricing.tsx:1]: Not even close

image

image

File: src/components/ui/button.tsx
Comment (bryanlundberg) [inline src/components/ui/button.tsx:1]: You should not commit files with only format changes.

**Review by bryanlundberg — CHANGES_REQUESTED** [review]
It's going to be hard to merge this PR.

I would recommend start again, because you over implemented the solution that was needed originally. Dialogs don't need a handler.

**Comment by adilkadivala** [comment]
Let me be close to issue,  

Here is what I understood..

when some one click on "upgrade to pro" button,  then pricing component should be poped up, and it should be close when click on "x" mark ,  and it will also closed when clicked outside of component.

allTasks

please look at here mine local machine 


https://github.com/user-attachments/assets/72543912-a7fa-4847-a777-46f80003888e

so for opening on Click I have passed props hanlder

 <LayoutApplicationNavigation>
          <Navigation />
          <Navigation handleDialog={handleDialog} />
        </LayoutApplicationNavigation>
        
        
now I'm waiting to you response for more clearity.

**Comment by adilkadivala** [comment]
oh sorry!! my bad.. 

I just realize that we didn't need to create any handler when we use shadcn ui,  sorry,  

let me solve it

**Comment by bryanlundberg** [comment]
I suggest start a new branch @adilkadivala 👌

**Pull request closed without merging**